frontend/src/views/Profile.tsx

```diff
@@ -1,16 +1,62 @@
 import { useParams } from 'react-router-dom';
+import { useEffect, useState } from 'react';
+import { useRecoilValue } from 'recoil';
+import { myIdState } from '../util/Recoils';
+import { socket } from '../util/Socket';
+import instance from '../util/Axios';
+import User from '../components/User/User';
+import ProfileMenuBar from '../components/Profile/MenuBar';
+import LadderProgressBar from '../components/Profile/LadderProgressBar';
+import Achievements from '../components/Profile/Achievements';
+import MatchHistoryList from '../components/Profile/MatchHistoryList';
+import WinLossTotalCounter from '../components/Profile/WinLossTotalCounter';
+import '../style/Profile/Profile.css';
 
+export interface profileDataForm {
+  ladder: number;
+  achievements: Array<{ id: number; title: string; about: string }>;
+  winLossTotal: Array<number>;
+  matchHistory: Array<{
+    winner: string;
+    loser: string;
+    score: Array<number>;
+    isRank: boolean;
+  }>;
+}
+
+/* TODO - uri의 id가 존재하지 않는 케이스 따로 처리 */
 function Profile() {
+  useEffect(() => {
+    socket.emit('currentUi', { ui: 'profile' });
+  }, []);
```

Comment:
connection 을 보장하기 위해 이렇게 고치면 더 나을 것 같아요~

```suggestion
  const [isConnected, setIsConnected] = useState(socket.connected);

  useCurrentUi(isConnected, setIsConnected, 'profile');
```

Reply:
으악 커넥션을 고려하지 않았네요............... 감사합니다!!!